handlers.go

```diff
@@ -161,6 +161,34 @@ func sha1Product(product string) string {
 	return product
 }
 
+// detect stub installers that pin the "DigiCert SHA2 Assured ID Code Signing CA" intermediate
+
+func isPre2024StubUserAgent(userAgent string) bool {
+	return "NSIS InetBgDL (Mozilla)" == userAgent
+}
+
+func pre2024Product(product string) string {
+	productParts := strings.SplitN(product, "-", 2)
+	if len(productParts) < 2 {
+		return product
+	}
+	if productParts[0] != "firefox" {
+		return product
+	}
+
+	fxPre2024LastNightly := "firefox-127.0a1"
```

Comment:
Let's move that to the top of the file.

Reply:
I can do that, but I'm curious why.  It's only used here, and in my mind moving it far from its use obscures things more than anything else.

Reply:
This is where constants live in golang files usually. I can quickly glance at the top of the file and see that the last nightly for "pre 2024" is _that_ version. In fact, I was confused about the last revision of the patch because I couldn't find this variable anywhere (well, until I explicitly searched for it and found it in the middle of the function).